Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -156,29 +160,34 @@ void SourceBufferPrivate::appendCompleted(bool parsingSucceeded, bool isEnded, F
 
     preAppendCompletedTask();
 
-    auto completionHandler = CompletionHandler<void()>([self = Ref { *this }, this, parsingSucceeded, isEnded]() {
-        if (!m_isAttached)
+    // Resolve the changes in TrackBuffers' buffered ranges
+    // into the SourceBuffer's buffered ranges
+    updateBufferedFromTrackBuffers(isEnded);
+
+    m_client->sourceBufferPrivateBufferedChanged(buffered(), [weakSelf = WeakPtr { *this }, this, parsingSucceeded] () mutable {
+        if (!weakSelf || !m_isAttached)
             return;
 
-        // Resolve the changes in TrackBuffers' buffered ranges
-        // into the SourceBuffer's buffered ranges
-        updateBufferedFromTrackBuffers(isEnded);
+        auto completionHandler = CompletionHandler<void()>([weakSelf = WTFMove(weakSelf), this, parsingSucceeded] {
+            if (!weakSelf || !m_isAttached)
+                return;
+
+            if (m_client) {
+                if (!m_didReceiveSampleErrored)
+                    m_client->sourceBufferPrivateAppendComplete(parsingSucceeded ? SourceBufferPrivateClient::AppendResult::Succeeded : SourceBufferPrivateClient::AppendResult::ParsingFailed);
+                m_client->sourceBufferPrivateReportExtraMemoryCost(totalTrackBufferSizeInBytes());
```

Comment:
Are we sure this is still valid after the call to sourceBufferPrivateAppendComplete?
Should we protecThis weakSelf?